jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Stream.java

```diff
@@ -886,12 +886,17 @@ private void notifyHeaders(HeadersFrame frame, Callback callback)
         }
     }
 
-    private void notifyDataAvailable()
+    private void notifyDataAvailable(boolean dispatch)
     {
         Listener listener = Objects.requireNonNullElse(getListener(), Listener.AUTO_DISCARD);
         try
         {
-            listener.onDataAvailable(this);
+            if (listener instanceof DispatchableListener dispatchableListener)
+                dispatchableListener.onDataAvailable(this, dispatch);
+            else if (dispatch && session.getEndPoint().getConnection() instanceof HTTP2Connection h2c)
```

Comment:
Wouldn't `session.getEndPoint().getConnection() instanceof HTTP2Connection` always be true?